src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -13701,6 +13699,9 @@ void Unit::SetInCombatState(bool PvP, Unit* enemy, uint32 duration)
 
         if (enemy)
         {
+            if (Creature* pCreature = ToCreature())
+                pCreature->UpdateLeashExtensionTime();
+
```

Comment:
```suggestion
            creature->UpdateLeashExtensionTime();

```
Creature's already been set up on L13692.

Reply:
Thank you